### DISCUSSION TITLE
uPnP Messages Filling Log?

### DISCUSSION THREAD

**Discussion by arrmo** [tagyoureit/nodejs-poolController#614] [General]
Hi,

Did some hunting around, but not finding this - likely me 😞. But ... I see the following filling my log - is this expected?
```
1|njsPC  | [9/17/2022, 4:15:28 PM] info: [4:15:28 PM] 192.168.2.64 GET /upnp.xml/device {}
1|njsPC  | [9/17/2022, 4:15:34 PM] info: [4:15:34 PM] 192.168.2.64 GET /upnp.xml/device {}
1|njsPC  | [9/17/2022, 4:15:40 PM] info: [4:15:40 PM] 192.168.2.64 GET /upnp.xml/device {}
```

Not a biggie, but curious of course 😆.

Thanks!

**Comment by rstrouse** [comment]
This is normal.  Any device that looks for upnp devices on your network will ask for these packets.   

We normally suppress them but but another device is looking for upnp devices on your network and tagging /device to the end of it incorrectly so it is showing in your log.  Do you have some HA service that is looking for /upnp.xml/device?  Fortunately, njsPC responds to this or it might be getting an error.

I suppose we could suppress that invalid request but I suspect something is tacking /device onto the end of the ssdp responses incorrectly.

**Comment by arrmo** [comment]
That makes sense! Hmmm ... so I fired up Wireshark, filtered by ssdp - and it seems like the `M-SEARCH` packets are coming from njsPC? I may be mis-reading that, very possibly!

Trying to see what is querying, like you say.

Thanks!

**Reply by rstrouse** [reply]
Yes njsPC will send out a udp M-SEARCH for any request.  As part of its response it will send a path to the device info which will be `http://<ipaddress>:port/upnp.xml`.   There are some that mistakenly add /device to the end of the request for a root device but that is incorrect.

If you type the above in your browser you will see the upnp device definition.

**Comment by arrmo** [comment]
That makes sense ... sort of, LOL! I'm likely mixing up the request and response. Where I'm a bit confused is that M-SEARCH is being sent by njsPC (right?), and from what I can find,
```
A client that wishes to discover available services on a network, uses method M-SEARCH.
Responses to such search requests are sent via unicast addressing to the originating address and port number of the multicast request.
```
From Wireshark, I see,
![image](https://user-images.githubusercontent.com/8756384/190880075-bd8bb150-c4dc-4cdf-b31c-8eea2843917d.png)

So it seems like njsPC is querying the network for devices? Or am I misunderstanding?

Thanks!

**Reply by rstrouse** [reply]
You are misunderstanding.  That M-SEARCH udp packet is a poll that is put out by anyone looking for upnp devices which happens when you register an ssdp server.   The response however will include a different packet that tells anyone who is interested where they can find the entire device definition through an http get call.  This will be in the upnpPath output which njsPC emits http://<ipaddress>/upnp.xml.   Something is then saying they are interested in the return and calling http://<ipaddress>/upnp.xml/device which is not correct.

I suspect it is probably one of the HASS or Hubitat interfaces.  Easy enough to fix and njsPC actually would have filtered the noise if /device wasn't appended to the end.  However, if you pull njsPC I have made the filter more aggressive so it will not show in the log.

**Reply by arrmo** [reply]
That makes sense, thanks! I don't have Hubitat on the network ... but perhaps HomeBridge? Will keep digging, just to understand (and find it if I can 😆). Appreciate the explanation!

PS: Still debugging the blown HW here, down to just the ICP not working => then will post back on the other discussion, to share info on parts, etc.
